### PR TITLE
docs: Add types reference

### LINKED ISSUE #1997: Move a few definitions from `dagger` to `core`
The following definitions should be moved from `dagger.io/dagger` to `dagger.io/dagger/core`:

- `#ImageConfig` (tightly coupled to our image-related core actions)
- `#HealthCheck` (same)
- `#Ref` (same)
- `#Scratch` (tightly coupled to our fs-related core actions)

I just hit the issue while developing a plan. I used `core.#Copy` and automatically assumed `core.#Scratch`. This is made worse by #1635 because if a definition doesn't exist (in this case `core.#Scratch`) it's not clear at all from the error. So you waste time debugging.

### PR DESCRIPTION
**Pull request by helderco**
Resolves #1997

About the _Data structures_ section, you shouldn't need the definitions directly, but it's useful to know which fields are there (e.g., `#ImageConfig` and `#HealtchCheck` when using the `config` field in `docker.#Set`).

Signed-off-by: Helder Correia

### CODE REVIEW

File: pkg/dagger.io/dagger/values.cue [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package dagger
```
Comment (TomChv) [inline pkg/dagger.io/dagger/values.cue:1]: Why not in `scratch.cue`?

File: docs/references/1234-dagger-types-reference.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+| `#CacheDir`    | [core/exec.cue](https://github.com/dagger/dagger/blob/v0.2.4/pkg/dagger.io/dagger/core/exec.cue) | A (best effort) persistent cache dir                  |
+| `#TempDir`     | [core/exec.cue](https://github.com/dagger/dagger/blob/v0.2.4/pkg/dagger.io/dagger/core/exec.cue) | A temporary directory for command execution           |
+
+### Related to container images
+
+| Definition     | File                                                                                             | Description                                           |
+| :------------- | :----------------------------------------------------------------------------------------------- | :---------------------------------------------------- |
+| `#ImageConfig` | [core/image.cue](https://github.com/dagger/dagger/blob/v0.2.4/pkg/dagger.io/dagger/image.cue)    | Container image config                                |
+| `#HealthCheck` | [core/image.cue](https://github.com/dagger/dagger/blob/v0.2.4/pkg/dagger.io/dagger/image.cue)    | Container health check                                |
```
Comment (TomChv) [inline docs/references/1234-dagger-types-reference.md:41]: What about other core actions?
`#WriteFile`, `#Pull` etc.. ?

**Review by TomChv — COMMENTED** [review]
LGTM, I just left questions

File: docs/references/1234-dagger-types-reference.md (hunk 2 above)
Comment (helderco) [reply docs/references/1234-dagger-types-reference.md:41]: Core actions are in another doc: https://docs.dagger.io/1222/core-actions-reference

**Review by aluzzardi — APPROVED** [review]

File: pkg/dagger.io/dagger/values.cue (hunk 1 above)
Comment (helderco) [reply pkg/dagger.io/dagger/values.cue:1]: See reasoning in https://github.com/dagger/dagger/issues/1997#issuecomment-1100039100

**Review by shykes — APPROVED** [review]
Love it.

**Pull request merged**